### PR TITLE
core-utils,contracts-bedrock,ci: OpNodeProvider + check

### PR DESCRIPTION
**Pull request by tynes**
**Description**

Implements a simple `OpNodeProvider` that can be used
to query the op-node. It is used as part of a ci check
against the devnet. It should add some more health related
things, now it just prints of rpc responses. This will
help to debug things.

### CODE REVIEW

File: packages/contracts-bedrock/tasks/check-op-node.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { task, types } from 'hardhat/config'
+import { OpNodeProvider } from '@eth-optimism/core-utils'
+
+// TODO(tynes): add in config validation
+task('check-op-node', 'Deposits funds onto L2.')
```
Comment (trianglesphere) [inline packages/contracts-bedrock/tasks/check-op-node.ts:5]: old task name?

Comment (tynes) [reply packages/contracts-bedrock/tasks/check-op-node.ts:5]: Ah the description, good catch

**Comment by tynes** [comment]
I don't think L2 ever finalizes based on the output of the CI here:

```
 "finalizedL2": {
    "hash": "0x0000000000000000000000000000000000000000000000000000000000000000",
    "number": {
      "type": "BigNumber",
      "hex": "0x00"
    },
    "parentHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
    "timestamp": {
      "type": "BigNumber",
      "hex": "0x00"
    },
    "l1Origin": {
      "hash": "0x0000000000000000000000000000000000000000000000000000000000000000",
      "number": {
        "type": "BigNumber",
        "hex": "0x00"
      }
    },
    "sequencerNumber": {
      "type": "BigNumber",
      "hex": "0x00"
    }
  }

```

**Comment by trianglesphere** [comment]
> I don't think L2 ever finalizes based on the output of the CI here:

That is correct. We don't set an L2 finalization block.

**Review by trianglesphere — APPROVED** [review]

**Review by mslipper — APPROVED** [review]

**Pull request merged**